### PR TITLE
Add googletest as a 3rdparty library

### PR DESCRIPTION
**Pull request by larroy**

### CODE REVIEW

**Comment by larroy** [comment]
This also helps when using CLion to develop, as the Gtest symbols are automatically resolved.

**Comment by larroy** [comment]
Please don't squash the commits when merging.

**Comment by larroy** [comment]
https://github.com/apache/incubator-mxnet/issues/9017

**Comment by larroy** [comment]
https://github.com/dmlc/mshadow/pull/313

**Comment by larroy** [comment]
https://github.com/dmlc/dmlc-core/pull/345

**Comment by cjolivier01** [comment]
This will probably need to be squashed, by the way :)

**Comment by larroy** [comment]
I will rebase at the end some commits, but squashing the addition of the gtest and the local changes makes the history more difficult to read or to diagnose problems when you bisect IMHO.

**Comment by larroy** [comment]
Please merge, this is badly needed.

**Comment by rahul003** [comment]
I agree, this makes it so much easier

**Comment by cjolivier01** [comment]
Why is appveyor failing? It has to pass to merge

**Comment by larroy** [comment]
@cjolivier01 weren't we disabling it today? Windows build is passing in Jenkins...

**Comment by cjolivier01** [comment]
Disabling appveyor?

**Comment by larroy** [comment]
@cjolivier01  right, it was voted and passed in the list, didn't you see?

**Comment by piiswrong** [comment]
please add a submodule. Don't copy paste the code

**Comment by larroy** [comment]
We discussed in the list to add it as a third party.... The library is very small, why is it better to have it as a submodule?

File: .gitmodules [hunk 1]
```diff
@@ -1,9 +1,9 @@
 [submodule "mshadow"]
 	path = mshadow
-	url = https://github.com/dmlc/mshadow.git
+	url = https://github.com/larroy/mshadow.git
 [submodule "dmlc-core"]
 	path = dmlc-core
-	url = https://github.com/dmlc/dmlc-core.git
+	url = https://github.com/larroy/dmlc-core.git
```
Comment (szha) [inline .gitmodules]: This is probably for testing. Leaving a note here just to make sure this change doesn't get accidentally merged.

**Comment by cjolivier01** [comment]
+1 to use as a submodule. This way our version can stay more current.

File: .gitmodules (hunk 1 above)
Comment (larroy) [reply .gitmodules]: Please ping me when the submodules are merged accordingly so we can rerun the build with the right submodule

**Comment by piiswrong** [comment]
Could you use the most recent release instead of current master for gtest?

**Comment by larroy** [comment]
@cjolivier01 can we merge this? I think it's important.

**Comment by cjolivier01** [comment]
Done

**Comment by larroy** [comment]
We had to merge two PRs to submodules before:

See: 

https://github.com/apache/incubator-mxnet/pull/9069

Then this PR restores the submodules.

File: .gitmodules (hunk 1 above)
Comment (szha) [reply .gitmodules]: 🤦‍♂️

**Pull request merged**